new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainInPV.java

```diff
@@ -0,0 +1,780 @@
+   *
+   * @param pvName name of the persistent volume to create
+   * @param domainUid domain UID
+   * @throws IOException when creating pv path fails
+   */
+  private void createPV(String pvName, String domainUid) throws IOException {
+    logger.info("creating persistent volume");
+
+    Path pvHostPath = Files.createDirectories(Paths.get(
+        PV_ROOT, this.getClass().getSimpleName(), pvName));
+    logger.info("Creating PV directory host path {0}", pvHostPath);
+    FileUtils.deleteDirectory(pvHostPath.toFile());
+    Files.createDirectories(pvHostPath);
+
+    V1PersistentVolume v1pv = new V1PersistentVolume()
+        .spec(new V1PersistentVolumeSpec()
+            .addAccessModesItem("ReadWriteMany")
+            .storageClassName("weblogic-domain-storage-class")
+            .volumeMode("Filesystem")
+            .putCapacityItem("storage", Quantity.fromString("5Gi"))
+            .persistentVolumeReclaimPolicy("Recycle")
+            .accessModes(Arrays.asList("ReadWriteMany"))
+            .hostPath(new V1HostPathVolumeSource()
+                .path(pvHostPath.toString())))
+        .metadata(new V1ObjectMeta()
+            .name(pvName)
+            .putLabelsItem("weblogic.resourceVersion", "domain-v2")
+            .putLabelsItem("weblogic.domainUid", domainUid));
+    boolean success = assertDoesNotThrow(() -> createPersistentVolume(v1pv),
+        "Failed to create persistent volume");
+    assertTrue(success, "PersistentVolume creation failed");
+  }
+
+  /**
+   * Create a persistent volume claim.
+   *
+   * @param pvName name of the persistent volume
+   * @param pvcName name of the persistent volume to create
+   * @param domainUid UID of the WebLogic domain
+   * @param namespace name of the namespace in which to create the persistent volume claim
+   */
+  private void createPVC(String pvName, String pvcName, String domainUid, String namespace) {
+    logger.info("creating persistent volume claim");
+
+    V1PersistentVolumeClaim v1pvc = new V1PersistentVolumeClaim()
+        .spec(new V1PersistentVolumeClaimSpec()
+            .addAccessModesItem("ReadWriteMany")
+            .storageClassName("weblogic-domain-storage-class")
+            .volumeName(pvName)
+            .resources(new V1ResourceRequirements()
+                .putRequestsItem("storage", Quantity.fromString("5Gi"))))
+        .metadata(new V1ObjectMeta()
+            .name(pvcName)
+            .namespace(namespace)
+            .putLabelsItem("weblogic.resourceVersion", "domain-v2")
+            .putLabelsItem("weblogic.domainUid", domainUid));
+
+    boolean success = assertDoesNotThrow(() -> createPersistentVolumeClaim(v1pvc),
+        "Failed to create persistent volume claim");
+    assertTrue(success, "PersistentVolumeClaim creation failed");
+  }
+
+  /**
+   * Create secret for docker credentials.
+   *
+   * @param namespace name of the namespace in which to create secret
+   */
+  private void createOCRRepoSecret(String namespace) {
```

Comment:
Is there a common utility method for to create secret for docker credentials. This is a very generic method across all test class.

Reply:
I created one in my PR.

Reply:
I have added a overloaded method in CommonTestUtils and using it inside this method.